packages/frontpage/app/(app)/item/[id]/_client.tsx

```diff
@@ -0,0 +1,176 @@
+"use client";
+import {
+  AlertDialogHeader,
+  AlertDialogFooter,
+} from "@/lib/components/ui/alert-dialog";
+import { Button } from "@/lib/components/ui/button";
+import { Textarea } from "@/lib/components/ui/textarea";
+import { SimpleTooltip } from "@/lib/components/ui/tooltip";
+import {
+  AlertDialog,
+  AlertDialogTrigger,
+  AlertDialogContent,
+  AlertDialogTitle,
+  AlertDialogDescription,
+  AlertDialogCancel,
+  AlertDialogAction,
+} from "@/lib/components/ui/alert-dialog";
+import {
+  ChatBubbleIcon,
+  ChevronUpIcon,
+  TrashIcon,
+} from "@radix-ui/react-icons";
+import { VariantProps, cva } from "class-variance-authority";
+import React, { useRef, useState } from "react";
+import { useToast } from "@/lib/components/ui/use-toast";
+
+const commentVariants = cva(undefined, {
+  variants: {
+    level: {
+      0: "",
+      1: "pl-8",
+      2: "pl-16",
+    },
+  },
+  defaultVariants: {
+    level: 0,
+  },
+});
+
+type CommentProps = VariantProps<typeof commentVariants> & {
+  id: string;
+  author: string;
+  comment: string;
+  createdAt: Date;
+};
+
+export function Comment({
+  id,
+  author,
+  comment,
+  level,
+  createdAt,
+}: CommentProps) {
+  const [showNewComment, setShowNewComment] = useState(false);
+  const commentRef = useRef<HTMLDivElement>(null);
+  const newCommentTextAreaRef = useRef<HTMLTextAreaElement>(null);
+  const { toast } = useToast();
+  return (
+    <>
```

Comment:
does this need to be wrapped with a fragment if everything is under `<article>` anyway?

Reply:
Nah can be removed, was there because I had the NewComment component as a sibling but moved it inside in the end.